pkg/apis/core/types_shoot.go

```diff
@@ -114,6 +114,11 @@ func (s *Shoot) GetProviderType() string {
 	return s.Spec.Provider.Type
 }
 
+// IsWorkerless checks if the shoot has zero workers.
+func (s *Shoot) IsWorkerless() bool {
+	return len(s.Spec.Provider.Workers) == 0
+}
```

Comment:
So far, we don't have methods on the API types themselves. I think, a helper function is more appropriate. You don't need to define it for the `v1alpha1` version.

Reply:
Adapted to use helper function instead of method in [4c9e1a5](https://github.com/gardener/gardener/pull/7828/commits/4c9e1a54c0b69e357a20a1f46621528589919f3e).